src/scenes/TaskDetailsView/components/DeletePrompt/index.tsx

```diff
@@ -0,0 +1,68 @@
+import {Modal} from "flowbite-react";
+import {useNavigate} from 'react-router-dom';
+import {useState} from "react";
+import {TaskService} from "../../../../services/taskService";
+import {Task} from "../../../../types/Task";
+
+interface DeletePromptProps {
+  show: boolean,
+  onClose: () => void,
+  task: Task
+}
+
+function DeletePrompt({ show, onClose, task }: DeletePromptProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [errorText, setErrorText] = useState<string | null>(null);
+
+  const navigate = useNavigate();
+  const redirectBack = () => {
+    navigate('/agent/tasks');
+  };
+
+  const handleSubmit = async () => {
+    setIsProcessing(true);
+    try {
+      await TaskService.deleteTask(task.id);
+      onClose();
+      redirectBack();
+    } catch (e: any) {
+      setErrorText(`Error: ${e.message}`);
+    } finally {
+      setIsProcessing(false);
+    }
+  }
+
+  return (
+    <Modal show={show} onClose={onClose} size="3xl">
+        <Modal.Header>
+            Delete Task
+        </Modal.Header>
+        <Modal.Body>
+            <div className="space-y-4">
+                <div className="p-6 text-center">
+                    {errorText && (
+                        <p className="text-sm text-red-500 font-normal dark:text-gray-400">
+                            {errorText}
+                        </p>
+                    )}
+                    <svg aria-hidden="true" className="mx-auto mb-4 w-14 h-14 text-gray-400 dark:text-gray-200" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
+                        <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z">
+                        </path>
+                    </svg>
```

Comment:
Let's use `react-icons` instead,
Pick one from this list https://react-icons.github.io/react-icons/